python/dgl/data/tree.py

```diff
@@ -125,6 +171,49 @@ def _rec_build(nid, node):
         ret.from_networkx(g, node_attrs=['x', 'y', 'mask'])
         return ret
 
+    def has_cache(self):
+        graph_path = os.path.join(self.save_path, self.mode + '_dgl_graph.bin')
+        ret = os.path.exists(graph_path)
+        if self.mode == 'train':
+            info_path = os.path.join(self.save_path, 'graph_info.pkl')
+            ret = ret and os.path.exists(info_path)
+        return ret
+
+    def save(self):
+        graph_path = os.path.join(self.save_path, self.mode + '_dgl_graph.bin')
+        save_graphs(graph_path, self.trees)
+        if self.mode == 'train':
+            info_path = os.path.join(self.save_path, 'info.pkl')
+            save_info(info_path, {'vocab': self.vocab, 'embed': self.pretrained_emb})
+
+    def load(self):
+        graph_path = os.path.join(self.save_path, self.mode + '_dgl_graph.bin')
+        self._trees = load_graphs(graph_path)
+        info_path = os.path.join(self.save_path, 'info.pkl')
+        if os.path.exists(info_path):
+            info = load_info(info_path)
+            self._vocab = info['vocab']
+            self._pretrained_emb = info['embed']
+
+    @property
+    def trees(self):
+        return self._trees
+
+    @property
+    def vocab(self):
+        r"""
```

Comment:
I'm still unclear about it: to which level do we need the docstrings? We might need an offline discussion